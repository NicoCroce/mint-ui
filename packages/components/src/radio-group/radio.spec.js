import * as React from "react";
import { cleanup, fireEvent, render} from "@testing-library/react";
import RadioGroup from "./index";

afterEach(cleanup);
describe("Radio group", () => {
    test("renders correctly", () => {
        const { asFragment } = render( 
            <RadioGroup data-testid="radio-group" value="item-4" onChange={(val) => {
                    // alert(val);
                }}>
                <RadioGroup.Button value="item-1">Item 1</RadioGroup.Button>
                <RadioGroup.Button value="item-2">Item 2</RadioGroup.Button>
                <RadioGroup.Button value="item-3">Item 3</RadioGroup.Button>
                <RadioGroup.Button value="item-4">Item 4</RadioGroup.Button>
            </RadioGroup>
        );  
        expect(asFragment()).toMatchSnapshot();
    });

    test("click works", (done) => {
        const handleClick = ((val)) => {
            expect(val).toBe("item-2");
            done();
        }
        const { getByTestId } = render(
            <RadioGroup data-testid="radio-test" value="item-4" onChange={handleClick}>
                <RadioGroup.Button value="item-1">Item 1</RadioGroup.Button>
                <RadioGroup.Button value="item-2">Item 2</RadioGroup.Button>
                <RadioGroup.Button value="item-3">Item 3</RadioGroup.Button>
                <RadioGroup.Button value="item-4">Item 4</RadioGroup.Button>
            </RadioGroup>
        );
        const component = getByTestId("radio-test");
        fireEvent.click(component.getElementsByTagName("label")[1]);
    });
})