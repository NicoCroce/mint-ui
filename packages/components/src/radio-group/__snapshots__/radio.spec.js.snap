// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio group renders correctly 1`] = `
<DocumentFragment>
  <div
    class="radio-group__Wrapper-sc-15twc87-0 bgTDsX"
    data-testid="radio-group"
  >
    <label
      class="radio-group__Label-sc-15twc87-1 cZnlzg radio-button"
    >
      Item 1
    </label>
    <label
      class="radio-group__Label-sc-15twc87-1 cZnlzg radio-button"
    >
      Item 2
    </label>
    <label
      class="radio-group__Label-sc-15twc87-1 cZnlzg radio-button"
    >
      Item 3
    </label>
    <label
      class="radio-group__Label-sc-15twc87-1 cfZFEP radio-button"
    >
      Item 4
    </label>
  </div>
</DocumentFragment>
`;
